bin/et

```diff
@@ -0,0 +1,62 @@
+#!/bin/bash
+#
+# Copyright 2013 The Flutter Authors. All rights reserved.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+
+set -e
+
+# Needed because if it is set, cd may print the path it changed to.
+unset CDPATH
+
+# On Mac OS, readlink -f doesn't work, so follow_links traverses the path one
+# link at a time, and then cds into the link destination and find out where it
+# ends up.
+#
+# The function is enclosed in a subshell to avoid changing the working directory
+# of the caller.
+function follow_links() (
+  cd -P "$(dirname -- "$1")"
+  file="$PWD/$(basename -- "$1")"
+  while [[ -h "$file" ]]; do
+    cd -P "$(dirname -- "$file")"
+    file="$(readlink -- "$file")"
+    cd -P "$(dirname -- "$file")"
+    file="$PWD/$(basename -- "$file")"
+  done
+  echo "$file"
+)
+
+SCRIPT_DIR=$(follow_links "$(dirname -- "${BASH_SOURCE[0]}")")
+FLUTTER_DIR="$(cd "$SCRIPT_DIR/.."; pwd -P)"
+
+case "$(uname -s)" in
+  Linux)
+    OS="linux"
+    ;;
+  Darwin)
+    OS="macos"
+    ;;
+  *)
+    echo "The host platform is not supported by this tool"
+    exit 1
+esac
+
+case "$(uname -m)" in
+  arm64)
+    CPU="arm64"
+    ;;
+  x86_64)
+    CPU="x64"
+    ;;
+  *)
+    echo "The host platform is not supported by this tool"
+    exit 1
+esac
+
+PLATFORM="${OS}-${CPU}"
+DART_SDK_DIR="${FLUTTER_DIR}/prebuilts/${PLATFORM}/dart-sdk"
+DART="${DART_SDK_DIR}/bin/dart"
+
+cd "${FLUTTER_DIR}/tools/engine_tool"
+"$DART" --disable-dart-dev bin/et.dart "$@"
```

Comment:
Should this run `dart pub get`?

If I run this on a clean machine I get:

```
loic@Pomelo:~/Code/engine/src/flutter$ ./bin/et
Error: Couldn't resolve the package 'engine_tool' in 'package:engine_tool/main.dart'.
bin/et.dart:5:8: Error: Not found: 'package:engine_tool/main.dart'
import 'package:engine_tool/main.dart' as m;
       ^
bin/et.dart:7:35: Error: Method not found: 'main'.
void main(List<String> args) => m.main(args);
                                  ^^^^
```

Reply:
Something like:

```
"$DART" pub upgrade > /dev/null
```

Reply:
I added a check for the `.dart_tool` directory. If it's not there, an error message directs to run `gclient sync -D`. I expect we'll use the tool to improve the engine repo setup and update flow later.

Reply:
Oh I see, didn't realize `gclient sync -D` handled this. Marking as resolved